### PR TITLE
Updated OAuth2 implementation

### PR DESCRIPTION
**Pull request by EconLQ**
removed GoogleTokenInfo and proceeded with the token validation process using GoogleIdTokenVerifier

### CODE REVIEW

**Comment by IhorKulishenko** [comment]
There is a little trap in method  `authenticateGoogle`  in particular in line 
`User user = (User) userService.userDetailsService().loadUserByUsername(tokenInfo.getEmail());`
if there there is no user with such email method `loadUserByUsername` doesn't return null but throws an exception, so instead of add new user to local db backend just returns message that user with email not found

File: src/main/java/com/gameshop/ecommerce/security/auth/service/AuthenticationService.java [hunk 1]
```diff
@@ -5,19 +5,81 @@
...
+            final var googleIdToken = new GoogleIdTokenVerifier.Builder(
+                    new NetHttpTransport(), new GsonFactory())
+                    .setAudience(List.of(googleClientId))
+                    .build()
+                    .verify(token);
+
+            if (googleIdToken != null) {
+                final var payload = googleIdToken.getPayload();
+                // check if user with this email already exists
+                User user = (User) userService.userDetailsService().loadUserByUsername(payload.getEmail());
```
Comment (IhorKulishenko) [inline src/main/java/com/gameshop/ecommerce/security/auth/service/AuthenticationService.java]: loadUserByUsername throws an exception if there is no user with the email in local db

**Review by IhorKulishenko — CHANGES_REQUESTED** [review]

**Comment by EconLQ** [comment]
intercepted exception in another method and returned empty optional:

```java
    private Optional<User> getLoadedUserByUsername(String email) {
        User user = null;
        try {
            user = (User) userService.userDetailsService().loadUserByUsername(email);
        } catch (UsernameNotFoundException e) {
            log.warn("User with email {} not found", email);
            return Optional.empty();
        }
        return Optional.of(user);
    }
```

updated logic inside the method `getUserFromGoogleIdToken(String token)`:

```java
              final var payload = googleIdToken.getPayload();
              // check if user with this email already exists
              Optional<User> user = getLoadedUserByUsername(payload.getEmail());
              if (user.isPresent()) {
                  return user;
              }
```

**Review by IhorKulishenko — CHANGES_REQUESTED** [review]
Pleas look at this updated logic and tell what you think, also I do some refactoring.

**Review by EconLQ — COMMENTED** [review]
I like the idea of implementing the UserDetailsService interface and overriding the loadUserByUsername method. The only suggestion is to add a log.warn() if `googleIdToken == null`, but that's a minor thing.

**Pull request merged**